src/main/java/com/amazon/ion/system/IonSystemBuilder.java

```diff
@@ -271,7 +276,111 @@ public final IonSystemBuilder withStreamCopyOptimized(boolean optimized)
         return b;
     }
 
+    //=========================================================================
+
+    /**
+     * Gets the text writer builder whose options will be used when building an
+     * {@link IonSystem}. By default, {@link IonTextWriterBuilder#standard()} will
+     * be used.
+     *
+     * @see #setIonTextWriterBuilder(IonTextWriterBuilder)
+     * @see #withIonTextWriterBuilder(IonTextWriterBuilder)
+     */
+    public final IonTextWriterBuilder getIonTextWriterBuilder() {
+        return textWriterBuilder;
+    }
+
+    /**
+     * Sets the text writer builder whose options will be used to use when building
+     * an {@link IonSystem}. The writer builder's catalog will never be used; the
+     * catalog provided to {@link #setCatalog(IonCatalog)} or
+     * {@link #withCatalog(IonCatalog)} will always be used instead.
```

Comment:
> the writer builder's catalog will never be used

This seems like it could be surprising behavior for users.

As long as one of the two catalog configurations (in `IonSystemBuilder` or the writer builder) is unset, then we should use the one that is set by the user. When both are set, we can specify that one takes precedence over the other. I think it is probably sensible to say that the more narrowly targeted catalog (i.e. in the writer builder) should take precedence over the catalog in `IonSystemBuilder`.

Reply:
It would be good to have unit tests that cover these configuration corner cases too.

Reply:
According to the constructor of [IonSystemLite](https://github.com/amazon-ion/ion-java/blob/master/src/main/java/com/amazon/ion/impl/lite/IonSystemLite.java#L87) behind this system builder, the catalog of all sub-builders (reader/writer) should be identical.

So, given this assertion, what could be a logical rule for defining the order of precedence? For example, if TWB, BWB and RB are all defined with a catalog?

Reply:
I added a test on the catalog definition, following the current behavior: the system catalog is taken priority over the writer catalog.